logproof/src/linear_relation.rs

```diff
@@ -1000,18 +1157,56 @@ mod test {
         // x^8 + 1
         let f = make_poly::<Q>(&[1, 0, 0, 0, 0, 0, 0, 0, 1]);
 
+        let d = f.degree();
+
         let t = &a * &s;
 
         let t_mod_f = (&t).scalar_rem(&f);
 
-        (a, s, t_mod_f, f)
+        let b = Matrix::from(
+            s_coeff
+                .iter()
+                .map(|row| {
+                    row.iter()
+                        .map(|coeffs| {
+                            let mut coeffs = coeffs.clone();
+                            coeffs.resize(d, 0);
+                            coeffs
+                                .into_iter()
+                                .map(|x| {
+                                    if x == 0 {
+                                        0
+                                    } else {
+                                        next_higher_power_of_two(x)
+                                    }
+                                })
+                                .collect::<Vec<u64>>()
+                        })
+                        .collect::<Vec<Vec<u64>>>()
+                })
+                .collect::<Vec<Vec<Vec<u64>>>>(),
+        );
```

Comment:
This seems bananas in terms of layers. If desired I could look into more comprehensible alternatives. Although 3 layer vecs may always look a bit complex.

Reply:
Also determined this was ok since it is test code.